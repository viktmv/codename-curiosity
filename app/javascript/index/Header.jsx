
import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'
import {
  BrowserRouter as Router,
  Route,
  Link,
  Switch
} from 'react-router-dom'

import SignUp from './SignUp'
import Login from './Login'
import LoL from './LoL'

class Header extends React.Component {
  render() {

    let loggedIn = this.props.user ? true : false
    let link = ''
      , text = ''

    if (loggedIn) {
      text = 'Logout'
      link = <Link to="/my-lists" className="my-lists-link">My Lists</Link>
    }
    else {
      text = 'Login'
      link = <Link to="/sign-up" className="singup-link">Sign Up</Link>
    }

    return (
      <header className="main-header">
        <div className="logo">Logo</div>
        <nav className="login-register-controls">
          <Router>
            <div>
              <ul>
                <ul>
                  <li><Link to="/login" className="login-link">{text}</Link></li>
                  <li>{link}</li>
                </ul>
              </ul>
              <Switch>
                <Route exact path="/login" render={this.renderLoginWithProps}/>
                <Route path="/sign-up" component={SignUp}/>
                <Route path="/my-lists" component={this.renderLoLWithProps}/>
              </Switch>
            </div>
          </Router>
        </nav>
      </header>
    )
  }

  renderLoginWithProps = () => {
    let {props} = this
    return <Login {...props} />
  }

  renderLoLWithProps = () => {
    let {props} = this
    return <LoL {...props} />
  }
}

export default Header
