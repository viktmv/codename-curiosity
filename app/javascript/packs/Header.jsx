
import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'
import {
  BrowserRouter as Router,
  Route,
  Link,
  Switch
} from 'react-router-dom'

const Login = () => (
  <div>
    <h2>Login</h2>
    <input name="email" type="email" placeholder="Email" />
    <input name="password" type="password" placeholder="password" />
    <button>Login</button>
  </div>
)

const SignUp = () => (
  <div>
    <h2>Sign-up</h2>
    <input name="email" type="email" placeholder="Email" />
    <input name="password" type="password" placeholder="password" />
    <input name="password-confirmation" type="password" placeholder="password" />
    <button>Sing- up</button>
  </div>
)

const LoL = () => (
  <div>
    <h2>My lists</h2>
    <ul>
      <li>1</li>
      <li>2</li>
      <li>3</li>
      <li>4</li>
      <li>5</li>
      <li>6</li>
      <li>7</li>
      <li>8</li>
      <li>9</li>
      <li>10</li>
      <li>11</li>
    </ul>
  </div>
)

class Header extends React.Component {
  render() {
    return (
      <header className="main-header">
        <nav className="login-register-controls">
          <div className="brand-logo">Logo</div>
            <Router>
              <div>
                <ul>
                  <ul>
                    <li><Link to="/login" className="login-link">Login</Link></li>
                    <li><Link to="/sign-up" className="singup-link">Sign Up</Link></li>
                    <li><Link to="/my-lists" className="my-lists-link">My Lists</Link></li>
                  </ul>
                </ul>
                <Route exact path="/login" component={Login}/>
                <Route path="/sign-up" component={SignUp}/>
                <Route path="/my-lists" component={LoL}/>
              </div>
            </Router>
        </nav>
      </header>
    )
  }
}

export default Header
