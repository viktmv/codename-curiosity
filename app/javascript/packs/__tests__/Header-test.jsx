import React from 'react';
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme';

import Header from '../Header.jsx';

test('Router renders the component when link is clicked', () => {
  const header = renderer.create(<Header></Header>)

  let tree = header.toJSON();
  expect(tree).toMatchSnapshot();

  // header.find('.login-link').simulate('click')

  // expect(header).toContain('<h2>Login</h2>')

});
