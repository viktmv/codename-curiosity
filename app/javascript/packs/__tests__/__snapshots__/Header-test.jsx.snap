// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router renders the component when link is clicked 1`] = `
<header
  className="main-header"
>
  <div
    className="logo"
  >
    Logo
  </div>
  <nav
    className="login-register-controls"
  >
    <div>
      <ul>
        <ul>
          <li>
            <a
              className="login-link"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li>
            <a
              className="singup-link"
              href="/sign-up"
              onClick={[Function]}
            >
              Sign Up
            </a>
          </li>
          <li>
            <a
              className="my-lists-link"
              href="/my-lists"
              onClick={[Function]}
            >
              My Lists
            </a>
          </li>
        </ul>
      </ul>
    </div>
  </nav>
</header>
`;
